op-devstack/dsl/proofs/fault_dispute_game.go

```diff
@@ -105,10 +124,25 @@ func (g *FaultDisputeGame) waitForClaim(timeout time.Duration, errorMsg string,
 		}
 		return false, nil
 	})
-	g.require.NoError(err, errorMsg)
-	// TODO(#15948) - Log GameData()
-	//if err != nil { // Avoid waiting time capturing game data when there's no error
-	//	g.require.NoErrorf(err, "%v\n%v", errorMsg, g.GameData(ctx))
-	//}
+	if err != nil { // Avoid waiting time capturing game data when there's no error
+		g.require.NoErrorf(err, "%v\n%v", errorMsg, g.GameData())
+	}
+
 	return matchClaimIdx, matchedClaim
 }
+
+func (g *FaultDisputeGame) GameData() string {
```

Comment:
This logic is based on the the split_game_helper [GameData() method](https://github.com/ethereum-optimism/optimism/blob/683588b177b593e75a9565b4da016dabd41f6c0f/op-e2e/e2eutils/disputegame/split_game_helper.go#L378-L395).

The only difference is that I dropped the `DescribePosition` logic [here](https://github.com/ethereum-optimism/optimism/blob/683588b177b593e75a9565b4da016dabd41f6c0f/op-e2e/e2eutils/disputegame/split_game_helper.go#L386) because it [requires a TraceProvider](https://github.com/ethereum-optimism/optimism/blob/683588b177b593e75a9565b4da016dabd41f6c0f/op-e2e/e2eutils/disputegame/super_game_helper.go#L39) and pulling all of that logic in is fairly complicated and probably not worth the effort right now.